### PR TITLE
[runtime] MonoError-ize mono_compile_method

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

**Comment by lambdageek** [comment]
build

File: mono/metadata/cominterop.c [hunk 1]
```diff
@@ -2115,7 +2115,8 @@ cominterop_get_ccw_checked (MonoObject* object, MonoClass* itf, MonoError *error
 			wrapper_method = mono_mb_create_method (mb, m.csig, m.csig->param_count + 16);
 			mono_cominterop_unlock ();
 
-			vtable [vtable_index--] = mono_compile_method (wrapper_method);
+			vtable [vtable_index--] = mono_compile_method_checked (wrapper_method, error);
+			return_val_if_nok (error, NULL);
```
Comment (kumpera) [inline mono/metadata/cominterop.c]: This leaks all the memory released right after the call.

File: mono/metadata/marshal.c [hunk 2]
```diff
@@ -395,7 +394,9 @@ mono_delegate_to_ftnptr (MonoDelegate *delegate)
 
 	wrapper = mono_marshal_get_managed_wrapper (method, klass, target_handle);
 
-	delegate->delegate_trampoline = mono_compile_method (wrapper);
+	delegate->delegate_trampoline = mono_compile_method_checked (wrapper, &error);
```
Comment (kumpera) [inline mono/metadata/marshal.c]: This leaks the weak handle.

File: mono/metadata/object.c [hunk 3]
```diff
@@ -7011,20 +7029,24 @@ mono_wait_handle_get_handle (MonoWaitHandle *handle)
...
 
 	RuntimeInvokeFunction runtime_invoke;
 
+	mono_error_init (error);
+
 	if (!domain->capture_context_runtime_invoke || !domain->capture_context_method) {
 		MonoMethod *method = mono_get_context_capture_method ();
 		MonoMethod *wrapper;
 		if (!method)
 			return NULL;
```
Comment (kumpera) [inline mono/metadata/object.c]: Must set the error object to something.

File: mono/metadata/remoting.c [hunk 4]
```diff
@@ -579,6 +582,14 @@ mono_marshal_emit_switch_domain (MonoMethodBuilder *mb)
 	mono_mb_emit_icall (mb, mono_marshal_set_domain_by_id);
 }
 
+gpointer mono_compile_method_icall (MonoMethod *method)
```
Comment (kumpera) [inline mono/metadata/remoting.c]: Formating, return type goes on its own line.

File: mono/metadata/object.c (hunk 3 above)
Comment (lambdageek) [inline mono/metadata/object.c]: There's no error condition to signal here: `method` could legitimately be `NULL`

File: mono/metadata/remoting.c [hunk 5]
```diff
@@ -81,6 +81,8 @@ static MonoMethod *method_rs_serialize, *method_rs_deserialize, *method_exc_fixe
 static MonoMethod *method_set_call_context, *method_needs_context_sink, *method_rs_serialize_exc;
 #endif
 
+static gpointer mono_compile_method_icall (MonoMethod *method);
```
Comment (lambdageek) [inline mono/metadata/remoting.c]: Another formatting thing.

**Comment by lambdageek** [comment]
Leaks fixed.

**Pull request merged**